### DISCUSSION TITLE
error: XDG_RUNTIME_DIR not set in the environment

### DISCUSSION THREAD

**Discussion by taxuezcy** [stella-cv/stella_vslam#539] [Q&A]
when run example ,the error is 

error: XDG_RUNTIME_DIR not set in the environment.
terminate called after throwing an instance of 'std::runtime_error'
  what():  Pangolin X11: Failed to open X display
Stack trace (most recent call last):
#18   Object "", at 0xffffffffffffffff, in 
#17   Source "/usr/include/c++/11/bits/shared_ptr_base.h", line 704, in _start [0x55dae1ec2134]
        702:       ~__shared_count() noexcept
        703:       {
      > 704: 	if (_M_pi != nullptr)
        705: 	  _M_pi->_M_release();
        706:       }
#16   Source "../csu/libc-start.c", line 392, in __libc_start_main_impl [0x7fb067d08e3f]
#15   Source "../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main [0x7fb067d08d8f]
#14   Source "/home/root4/stella_vslam/example/run_video_slam.cc", line 295, in main [0x55dae1ec18a7]
        292:     // run tracking
        293:     int ret;
        294:     if (slam->get_camera()->setup_type_ == stella_vslam::camera::setup_type_t::Monocular) {
      > 295:         ret = mono_tracking(slam,
        296:                             cfg,
        297:                             video_file_path->value(),
        298:                             mask_img_path->value(),
#13   Source "/home/root4/stella_vslam/example/run_video_slam.cc", line 141, in mono_tracking [0x55dae1ec3cd1]
        138:     if (!disable_gui) {
        139:         // run the viewer in the current thread
        140: #ifdef USE_PANGOLIN_VIEWER
      > 141:         viewer.run();
        142: #elif USE_SOCKET_PUBLISHER
        143:         publisher.run();
        144: #endif
#12   Source "/home/root4/stella_vslam/src/pangolin_viewer/viewer.cc", line 54, in run [0x7fb069f39703]
         51: void viewer::run() {
         52:     is_terminated_ = false;
         53: 
      >  54:     pangolin::CreateWindowAndBind(map_viewer_name_, 1024, 768);
         55: 
         56:     glEnable(GL_BLEND);
         57:     glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);

**Comment by Skwangles** [comment]
First make sure you have GLEW and a supported pangolin version installed
https://stella-cv.readthedocs.io/en/latest/installation.html#requirements-for-pangolinviewer

Second, try googling the error and trying some of those fixes
e.g. https://www.reddit.com/r/7daystodie/comments/14bvbfa/error_xdg_runtime_dir_not_set_in_the_environment/
https://askubuntu.com/questions/456689/error-xdg-runtime-dir-not-set-in-the-environment-when-attempting-to-run-naut, https://github.com/swaywm/sway/issues/7202
